scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -160,11 +160,16 @@ def make_env_mach_specific_file(self, shell, case):
         with open(filename, "w") as fd:
             fd.write("\n".join(lines))
 
-    def _load_envs(self, envs_to_set):
+    # Private API
+
+    def _load_envs(self, envs_to_set, verbose=False):
         for env_name, env_value in envs_to_set:
             os.environ[env_name] = "" if env_value is None else env_value
-
-    # Private API
+            if verbose:
+                if env_value is not None:
+                    logger.warning("Setting Environment {}={}".format(env_name, env_value))
+                else:
+                    logger.warning("Unsetting Environment {}".format(env_name))
```

Comment:
This warning is not accurate. Setting an env var to "" is not the same as unsetting it. I'd prefer this block to match the block in load_module_modules:

```
            if verbose:
                logger.warning("Setting Environment {}={}".format(env_name, env_value))
            else:
                logger.debug("Setting Environment {}={}".format(env_name, env_value))
```

Reply:
I'm now actually unsetting the env var if it exists so the statement is correct.